cmd/kubectl-gadget/undeploy.go

```diff
@@ -150,9 +177,36 @@ again:
 		errs = append(errs, fmt.Sprintf("failed to remove \"gadget\" namespace: %s", err))
 	}
 
+	if undeployWait && watcher != nil {
+		ch := watcher.ResultChan()
+		defer watcher.Stop()
+
+		timer := time.NewTimer(time.Duration(timeout) * time.Second)
+
+	loop:
+		for {
```

Comment:
Isn't there an easy way to avoid busy waiting?
Something like what is done in `waitForCondition()` but for NameSpace (hoping the API will be easier to handle)?

Reply:
It's not busy waiting, notice that there is a channel read on L189. I used a loop in case we receive multiple events for the `gadget` namespace before the delete one. Actually this code is adapted from [untilWithouRetry](https://github.com/kinvolk/inspektor-gadget/blob/8c5811050cb8cf9dcd588b520d5f36e0a44da055/cmd/kubectl-gadget/utils/trace.go#L414), let me see if we can reuse that function there.

Reply:
I reworked the PR to use `watchtools.Until()`, so I don't have to implement this logic by hand.